src/map/Map.js

```diff
@@ -511,15 +518,20 @@ export var Map = Evented.extend({
 		return this;
 	},
 
-	// @method panInside(latlng: LatLng, options?: options): this
+	// @method panInside(latlng: LatLng, options?: padding options): this
 	// Pans the map the minimum amount to make the `latlng` visible. Use
-	// `padding`, `paddingTopLeft` and `paddingTopRight` options to fit
-	// the display to more restricted bounds, like [`fitBounds`](#map-fitbounds).
+	// padding options to fit the display to more restricted bounds.
 	// If `latlng` is already within the (optionally padded) display bounds,
 	// the map will not be panned.
 	panInside: function (latlng, options) {
 		options = options || {};
 
+		if ((options.padding && options.padding.length !== 2) ||
+			(options.paddingTopLeft && options.paddingTopLeft.length !== 2) ||
+			(options.paddingBottomRight && options.paddingBottomRight.length !== 2)) {
+			throw new Error('Padding must be a point type.');
+		}
+
 		var paddingTL = toPoint(options.paddingTopLeft || options.padding || [0, 0]),

```

Comment:
The best solution might be including those checks into `toPoint` function.
What do you think?

Reply:
As for me - I believe that it would be most clean solution, with much more positive impact, because it is able to catch Point misuse in every place, not only in padding options.
But I have some concerns about (1) compatibility and (2) performance.
1. Perhaps not an issue. Who would want to call `toPoint(50)` intentionally?
2. We need to make sure that `toPoint` is not used in any hot path.

Reply:
I thought about this a few days ago, but I discounted it because I've used `toPoint` in the past on quite large datasets and I was worried the additional checks could affect performance in such cases. I think putting validation on padding options and improving the documentation as we've done here is a good start, and maybe `toPoint` can be revisited in the future.

Reply:
> I discounted it because I've used `toPoint` in the past on quite large datasets and I was worried the additional checks could affect performance in such cases.

Still, I believe that it is very important - to catch bugs early, so I've added this case to this discussion https://github.com/Leaflet/Leaflet/issues/7128#issuecomment-760765116